otter/cloud_client.py

```diff
@@ -135,10 +143,38 @@ def __init__(self, effect, tenant_id):
         self.tenant_id = tenant_id
 
 
+def _add_service_error_parsing(parser, request_func):
+    """
+    If a parser for API errors is found, use it to parse service API errors.
+
+    :param callable parser: A function which takes a tuple of exc info, to
+        be called when the request function raises an :class:`APIError`.
+    :param callable request_func: The request function to decorate
+
+    If there is no parser, does nothing.
+    """
+    if parser is None:
+        return request_func
+
+    def call_parser(excinfo):
+        # ensures that if the parser doesn't raise another exception,
+        # the original exception is raised
+        parser(excinfo)
+        six.reraise(*excinfo)
+
+    @wraps(request_func)
+    def request(*args, **kwargs):
+        return request_func(*args, **kwargs).on(
+            error=catch(APIError, call_parser))
+
+    return request
+
+
 def concretize_service_request(
         authenticator, log, service_configs,
         tenant_id,
-        service_request):
+        service_request,
+        service_error_parsers=None):
```

Comment:
The callers should definitely not provide dicts.  This was mostly a mechanism to provide incremental functionality.  I wanted to not touch the steps at all until the parsers were done, and also implement parsers one by one.  

This was mostly for easy splitting of PRs and also test injection - I wanted to make sure the mechanism by which the parsers were called were correct before actually implementing a parser.  I had figured that eventually I'd make the default be the correct dict.

I'm fine with getting rid of it later - how do you think we should proceed?  Should I just merge my CLB parser into this PR, and use that to test that if there is a parser for a service type, errors are parsed, and if not, errors are not parsed?